src/test/java/seedu/sudohr/logic/commands/leave/AddEmployeeToLeaveCommandTest.java

```diff
@@ -0,0 +1,432 @@
+import seedu.sudohr.logic.commands.CommandResult;
+import seedu.sudohr.logic.commands.exceptions.CommandException;
+import seedu.sudohr.model.Model;
+import seedu.sudohr.model.ReadOnlySudoHr;
+import seedu.sudohr.model.ReadOnlyUserPrefs;
+import seedu.sudohr.model.SudoHr;
+import seedu.sudohr.model.department.Department;
+import seedu.sudohr.model.department.DepartmentName;
+import seedu.sudohr.model.employee.Employee;
+import seedu.sudohr.model.employee.Id;
+import seedu.sudohr.model.leave.Leave;
+import seedu.sudohr.model.leave.LeaveDate;
+import seedu.sudohr.testutil.TypicalEmployees;
+import seedu.sudohr.testutil.TypicalLeave;
+
+/**
+ * Adds a employee using it's displayed index to a specific leave using it's
+ * displayed index in sudohr book.
+ */
+public class AddEmployeeToLeaveCommandTest {
+
+    @Test
+    public void execute_employeeAcceptedByLeave_addSuccessful() throws CommandException {
+
+        ModelStubAcceptingEmployeeAdded modelStub = new ModelStubAcceptingEmployeeAdded();
+        modelStub.addEmployee(TypicalEmployees.ALICE);
+        CommandResult commandResult = new AddEmployeeToLeaveCommand(
+                TypicalEmployees.ALICE_ID,
+                new LeaveDate(LocalDate.parse(VALID_LEAVE_DATE_LEAVE_TYPE_1))).execute(modelStub);
+
+        assertEquals(String.format(AddEmployeeToLeaveCommand.MESSAGE_ADD_LEAVE_SUCCESS,
+                TypicalEmployees.ALICE, TypicalLeave.LEAVE_TYPE_1),
+                commandResult.getFeedbackToUser());
+
+        assertTrue(modelStub.sudoHr.getLeave(new Leave(new LeaveDate(LocalDate.parse(
+                VALID_LEAVE_DATE_LEAVE_TYPE_1))))
+                .hasEmployee(TypicalEmployees.ALICE));
+    }
+
+    // handle adding to leave objects that already exists
+
+    @Test
+    public void execute_employeeAcceptedByLeaveAndPreviousEmployeeExists_addSuccessful() throws CommandException {
+
+        ModelStubAcceptingEmployeeAdded modelStub = new ModelStubAcceptingEmployeeAdded();
+        modelStub.addEmployee(TypicalEmployees.ALICE);
+        modelStub.addEmployee(TypicalEmployees.BENSON);
+
+        new AddEmployeeToLeaveCommand(
+                TypicalEmployees.ALICE_ID,
+                new LeaveDate(LocalDate.parse(VALID_LEAVE_DATE_LEAVE_TYPE_1))).execute(modelStub);
+
+        CommandResult commandResult = new AddEmployeeToLeaveCommand(
+                TypicalEmployees.BENSON_ID,
+                new LeaveDate(LocalDate.parse(VALID_LEAVE_DATE_LEAVE_TYPE_1))).execute(modelStub);
+
+        assertEquals(String.format(AddEmployeeToLeaveCommand.MESSAGE_ADD_LEAVE_SUCCESS,
+                TypicalEmployees.BENSON, TypicalLeave.LEAVE_TYPE_1),
+                commandResult.getFeedbackToUser());
+
+        assertTrue(modelStub.sudoHr.getLeave(new Leave(new LeaveDate(LocalDate.parse(
+                VALID_LEAVE_DATE_LEAVE_TYPE_1))))
+                .hasEmployee(TypicalEmployees.ALICE));
+    }
+
+    // handle duplicate employee in Leave
```

Comment:
can consider splitting test cases into duplicate ID and duplicate employee (same object reference) though they are semantically the same in our definition. Just to be comprehensive. Minor point tho.

Reply:
Currently add Employee to Leave exclusive uses EID so I think its fine? But since u mentioned this I just added test case with duplicated Id